### PR TITLE
ticdc/config: add feature debug configuration

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve? 
Add a new section into the ticdc configuration file named "debug" to hide some feature configs
Strict configuration format checking in this section will be ignored.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: pkg/cmd/server/server.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func (o *options) complete(cmd *cobra.Command) error {
 	cfg := config.GetDefaultServerConfig()
 
 	if len(o.serverConfigFilePath) > 0 {
-		if err := util.StrictDecodeFile(o.serverConfigFilePath, "TiCDC server", cfg); err != nil {
+		if err := util.StrictDecodeFile(o.serverConfigFilePath, "TiCDC server", cfg, "debug"); err != nil {
```
Comment (overvenus) [inline pkg/cmd/server/server.go:147]: Please add some comments about "debug".

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: pkg/cmd/server/server.go (hunk 1 above)
Comment (sdojjy) [reply pkg/cmd/server/server.go]: done

**Review by overvenus — APPROVED** [review]
LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**